src/main/java/de/tum/cit/aet/artemis/iris/web/open/PublicPyrisStatusUpdateResource.java

```diff
@@ -171,7 +183,58 @@ public ResponseEntity<Void> setStatusOfIngestionJob(@PathVariable String runId,
         }
 
         pyrisStatusUpdateService.handleStatusUpdate(ingestionWebhookJob, statusUpdateDTO);
-
         return ResponseEntity.ok().build();
     }
+
+    /**
+     * Retrieves the overall ingestion state of a lecture by communicating with Pyris.
+     *
+     * <p>
+     * This method sends a GET request to the external Pyris service to fetch the current ingestion
+     * state of all lectures in a course, identified by its `lectureId`. The ingestion state can be aggregated from
+     * multiple lecture units or can reflect the overall status of the lecture ingestion process.
+     * </p>
+     *
+     * @param courseId the ID of the lecture for which the ingestion state is being requested
+     * @return a {@link ResponseEntity} containing the {@link IngestionState} of the lecture,
+     */
+    @GetMapping("courses/{courseId}/lectures/ingestion-state")
+    @EnforceNothing
```

Comment:
If I understand this file `PublicPyrisStatusUpdateResource` correctly, it is meant to be solely used for communication from Pyris to Artemis. As the JavaDoc in the header recommends, you should always use `getAndAuthenticateJobFromHeaderElseThrow` to ensue the caller is a Pyris service and uses the right authentication token.

When I understand this PR correctly, the two additional methods added in this PR are meant for communication between the Artemis client and the Artemis server. Please move them into a different RestController and make sure to use appropriate authorization handling. While both PRs do not really reveal secrets, it is not a good idea to offer them to any user in an unauthorized way.